.github/workflows/run_periodic_tests.yml

```
@@ -0,0 +1,128 @@
+# Run all unit tests and integration tests for all Python versions 
+# and platforms at 3am UTC every day and on PRs to the main branch
+name: PyBaMM
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: 
+    - main
+
+  # Run everyday at 3 am UTC
+  schedule:
+    - cron: "0 3 * * *"
+
+jobs:
+  pre_job:
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          # All of these options are optional, so you can remove them if you are happy with the defaults
+          concurrent_skipping: "never"
+          cancel_others: "true"
+          paths_ignore: '["**/README.md"]'
+
+  style:
+    needs: pre_job
+    if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
```

`3.11` is faster. I am not sure if we will get speed up here, but we can try.
```suggestion
          python-version: 3.11
```